extensions/positron-assistant/src/openai-fetch-utils.ts

```diff
@@ -120,17 +266,15 @@ function transformServerSentEvents(text: string, providerName: string): string {
 			try {
 				const jsonStr = line.slice(6); // Remove 'data: ' prefix
 				const data = JSON.parse(jsonStr);
-
-				// Fix empty role fields in delta objects within choices array
-				if (data.choices && Array.isArray(data.choices)) {
-					for (const choice of data.choices) {
-						if (choice.delta && typeof choice.delta === 'object' && choice.delta.role === '') {
-							choice.delta.role = 'assistant';
-						}
-					}
+				// Check if it's a possibly broken chunk and fix it
+				// Otherwise, keep the original line
+				if (isPossiblyBrokenChatCompletionChunk(data)) {
```

Comment:
Just a follow on to the note above about the type and type guard mismatch: if `isPossiblyBrokenChatCompletionChunk()` actually matched the `PossiblyBrokenChatCompletionChunk` type, then it would always enter this `if` block and never go into the `else`. (Assuming that `data` is a JS object, as opposed to something like a string or null.)